### PR TITLE
Use setuptools_scm for package versions

### PR DESCRIPTION
**Pull request by dweindl**
Removes some extra `build-system.requires` entries.

So far, repository versions were date-based (well, there was only one, v20230717). The Python package had a constant version number `0.0.0a1`.
To work well with the default setuptools_scm schema, future releases should ideally use `vYYYY.MM.DD` tags.
The Python package will then also have the same version as the benchmark collection as such.

### CODE REVIEW

**Review by dilpath — APPROVED** [review]
Thanks, should we delete the fallback version and make a GitHub release immediately after merging this, with the appropriate version format?

File: src/python/benchmark_models_petab/__init__.py [hunk 1]
```diff
@@ -6,5 +6,11 @@
 
 from .base import get_problem, get_problem_yaml_path
 from .C import MODEL_DIRS, MODELS, MODELS_DIR
-from .version import __version__
 from .overview import get_overview_df
+from importlib.metadata import PackageNotFoundError, version
+
+try:
+    __version__ = version("sbmlmath")
```
Comment (dilpath) [inline src/python/benchmark_models_petab/__init__.py:13]:
```suggestion
    __version__ = version("benchmark_collection_petab")
```

Comment (dweindl) [reply src/python/benchmark_models_petab/__init__.py]: :see_no_evil:

**Comment by dweindl** [comment]
> Thanks, should we delete the fallback version and make a GitHub release immediately after merging this, with the appropriate version format?

Good to make a release, but no need to remove the fallback version. This will just be ignored then.

**Pull request merged**